include/fastdds/rtps/transport/TCPTransportDescriptor.h

```diff
@@ -54,132 +140,148 @@ typedef struct TCPTransportDescriptor : public SocketTransportDescriptor
+                   this->rsa_private_key_file == t.rsa_private_key_file &&
+                   this->handshake_role == t.handshake_role);
         }
 
-        TLSConfig(const TLSConfig& t)
-            : password(t.password)
-            , options(t.options)
-            , cert_chain_file(t.cert_chain_file)
-            , private_key_file(t.private_key_file)
-            , tmp_dh_file(t.tmp_dh_file)
-            , verify_file(t.verify_file)
-            , verify_mode(t.verify_mode)
-            , verify_paths(t.verify_paths)
-            , default_verify_path(t.default_verify_path)
-            , verify_depth(t.verify_depth)
-            , rsa_private_key_file(t.rsa_private_key_file)
-            , handshake_role(t.handshake_role)
-        {
-        }
-
-        TLSConfig(TLSConfig&& t)
-            : password(std::move(t.password))
-            , options(std::move(t.options))
-            , cert_chain_file(std::move(t.cert_chain_file))
-            , private_key_file(std::move(t.private_key_file))
-            , tmp_dh_file(std::move(t.tmp_dh_file))
-            , verify_file(std::move(t.verify_file))
-            , verify_mode(std::move(t.verify_mode))
-            , verify_paths(std::move(t.verify_paths))
-            , default_verify_path(std::move(t.default_verify_path))
-            , verify_depth(std::move(t.verify_depth))
-            , rsa_private_key_file(std::move(t.rsa_private_key_file))
-            , handshake_role(std::move(t.handshake_role))
-        {
-        }
-
-        TLSConfig& operator=(const TLSConfig& t)
-        {
-            password = t.password;
-            options = t.options;
-            cert_chain_file = t.cert_chain_file;
-            private_key_file = t.private_key_file;
-            tmp_dh_file = t.tmp_dh_file;
-            verify_file = t.verify_file;
-            verify_mode = t.verify_mode;
-            verify_paths = t.verify_paths;
-            default_verify_path = t.default_verify_path;
-            verify_depth = t.verify_depth;
-            rsa_private_key_file = t.rsa_private_key_file;
-            handshake_role = t.handshake_role;
-
-            return *this;
-        }
     };
 
+    //! List of ports to listen as server
     std::vector<uint16_t> listening_ports;
+    //! Frequency of RTCP keep alive requests (ms)
     uint32_t keep_alive_frequency_ms;
+    //! Time since sending the last keep alive request to consider a connection as broken (ms)
     uint32_t keep_alive_timeout_ms;
+    //! Maximum number of logical ports to try during RTCP negotiation
     uint16_t max_logical_port;
+    //! Maximum number of logical ports per request to try during RTCP negotiation
     uint16_t logical_port_range;
+    //! Increment between logical ports to try during RTCP negotiation
     uint16_t logical_port_increment;
+    // TODO(jlbueno): Remove v3.0
     uint32_t tcp_negotiation_timeout;
+    //! Enables the TCP_NODELAY socket option
     bool enable_tcp_nodelay;
+    // TODO(jlbueno): Remove v3.0
```

Comment:
Instead of leaving those `TODO`... should we deprecate `tcp_negotiation_timeout` and `wait_for_tcp_negotiation`?

Reply:
It looks like if you deprecate a public attribute from a class that is still in use, a warning will arise every time a constructor of such class is compiled (even when the attribute is not called in the constructor or in any other part of the code).
Regarding the c++ documentation, it should be possible to deprecate a variable (https://en.cppreference.com/w/cpp/language/attributes/deprecated) but it appears that, as it is default constructed in any constructor, an attribute cannot be deprecated.

I cannot think of any other solution than leaving a comment (or macro with comment `FASTDDS_DEPRECATED_UNTIL__TODO`).

Reply:
Finally a new Macro `FASTDDS_TODO` has been implemented, and added as a reminder to erase this two attributes bore v3.0.